src/libc-wrappers/libc-wrappers.c

```diff
@@ -15,7 +15,153 @@
  */
 
 
+#include <pthread.h>
 #include <signal.h>
+#include <link.h>
+
+
+/*
+ * __libc_start_main
+ */
+
+
+struct startup_info
+{
+  void *sda_base;
+  int (*main) (int, char **, char **, void *);
+  int (*init) (int, char **, char **, void *);
+  void (*fini) (void);
+};
+
+extern int __libc_start_main (int argc,
+                              char **argv,
+                              char **ev,
+                              ElfW (auxv_t) * auxvec,
+                              void (*rtld_fini) (void),
+                              struct startup_info * stinfo,
+                              char **stack_on_entry);
+
+
+#if defined __aarch64__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.17");
+#elif defined __arm__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.4");
+#elif defined __i386__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.0");
+#elif defined __powerpc64__ && _CALL_ELF == 2 /* ppc64le */
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.17");
+#elif defined __s390x__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.2");
+#elif defined __x86_64__
+__asm__(".symver __libc_start_main,__libc_start_main@GLIBC_2.2.5");
+#else
+#error "Please specify symbol version for __libc_start_main"
+#endif
+
+
+int
+__wrap___libc_start_main (int argc,
+                          char **argv,
+                          char **ev,
+                          ElfW (auxv_t) * auxvec,
+                          void (*rtld_fini) (void),
+                          struct startup_info * stinfo,
+                          char **stack_on_entry)
+{
+  return __libc_start_main (argc, argv, ev, auxvec, rtld_fini, stinfo, stack_on_entry);
+}
+
+
+/*
+ * pthread_attr_getstacksize
+ */
+
+
+#if defined __aarch64__
+__asm__(".symver pthread_attr_getstacksize,pthread_attr_getstacksize@GLIBC_2.17");
+#elif defined __arm__
+__asm__(".symver pthread_attr_getstacksize,pthread_attr_getstacksize@GLIBC_2.4");
+#elif defined __i386__
+__asm__(".symver pthread_attr_getstacksize,pthread_attr_getstacksize@GLIBC_2.1");
+#elif defined __powerpc64__ && _CALL_ELF == 2 /* ppc64le */
+__asm__(".symver pthread_attr_getstacksize,pthread_attr_getstacksize@GLIBC_2.17");
+#elif defined __s390x__
+__asm__(".symver pthread_attr_getstacksize,pthread_attr_getstacksize@GLIBC_2.2");
+#elif defined __x86_64__
+__asm__(".symver pthread_attr_getstacksize,pthread_attr_getstacksize@GLIBC_2.2.5");
+#else
+#error "Please specify symbol version for pthread_attr_getstacksize"
+#endif
+
+
+int
+__wrap_pthread_attr_getstacksize (const pthread_attr_t *attr, size_t *stacksize)
+{
+  return pthread_attr_getstacksize (attr, stacksize);
+}
+
+
+/*
+ * pthread_create
+ */
+
+
+#if defined __aarch64__
+__asm__(".symver pthread_create,pthread_create@GLIBC_2.17");
+#elif defined __arm__
+__asm__(".symver pthread_create,pthread_create@GLIBC_2.4");
+#elif defined __i386__
+__asm__(".symver pthread_create,pthread_create@GLIBC_2.0");
```

Comment:
Isn't GLIBC_2.1 the latest version of the `pthread_create` symbol on i386 that we can use?

I looked at the corresponding [glibc commit](https://sourceware.org/git/?p=glibc.git;a=commit;h=f47f1d91af985a9028fb399da21eab460d887a15), and looked inside the latest Fedora 33 binary for `i686` with:
```
$ rpm2cpio ./toolbox-0.0.99.2-1.fc33.i686.rpm | cpio -idmv
...
...
$ objdump -T ./usr/bin/toolbox | grep pthread_create
00000000      DF *UND*	00000000  GLIBC_2.1   pthread_create
002b4980 g    DF .text	000000cc  Base        _cgo_try_pthread_create
```

Reply:
I took the liberty to update the branch.